src/runtime_src/core/common/api/xrt_error.cpp

```diff
@@ -205,6 +206,19 @@ class error_impl
 public:
   error_impl(const xrt_core::device* device, xrtErrorClass ecl)
   {
+    //Code for new format; Binary array of error structs in sysfs
+    try {
+      auto buf = xrt_core::device_query<xrt_core::query::xocl_errors>(device);
```

Comment:
Much cleaner and leaner now.  Thanks!

xocl_errors is not a great name for a query unless this is temporary until error has moved to new format on zocl also.
At the very least, can we keep the key of the query_request the same for both xocl_errors and error?

Reply:
Yes this is temporary. Will change to "errors" later as it will contain multiple errors.